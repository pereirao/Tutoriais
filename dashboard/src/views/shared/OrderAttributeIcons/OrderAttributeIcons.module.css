.attributeCell {
  display: flex;
}

.attributeIcon {
  height: 24px;
  width: 24px;
  margin-right: 8px;
}