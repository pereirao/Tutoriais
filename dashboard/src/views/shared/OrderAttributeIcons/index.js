//@flow

import React from 'react';
import style_sheet from './OrderAttributeIcons.module.css';
import MBIcon from '../../elements/MBIcon';
import I18n from '../../../localization';


type OrderAttributeIconsProps = { attributes: Array<string>, iconClassName?: Object, showSubstitutionOk: boolean };
const OrderAttributeIcons = ({attributes, iconClassName, showSubstitutionOk}: OrderAttributeIconsProps) => (
  <div className={style_sheet.attributeCell}>
    {attributes.filter(attribute => (
      (!showSubstitutionOk && attribute !== 'allow_substitution') || showSubstitutionOk
    )).map(attribute => (
      <MBIcon
        key={`${attribute}icon`}
        color={attribute === 'engraving' ? 'mb_white' : 'mb_medium_grey'}
        inline
        icon={attribute}
        strokeColor={attribute === 'engraving' ? '#A0A0A0' : 'none'}
        strokeWidth={attribute === 'engraving' ? '30' : 'none'}
        className={iconClassName || style_sheet.attributeIcon}
        title={I18n.t(`accessibility.icon.${attribute}`)}/>
    ))}
  </div>
);

export default OrderAttributeIcons;
