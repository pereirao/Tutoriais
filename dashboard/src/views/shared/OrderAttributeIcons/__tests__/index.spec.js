import React from 'react';

import OrderAttributeIcons from '../index';

it('renders correctly', () => {
  const attributes = ['gift', 'vip', 'scheduled'];
  expect(
    shallow(<OrderAttributeIcons attributes={attributes} />)
  ).toMatchSnapshot();
});
