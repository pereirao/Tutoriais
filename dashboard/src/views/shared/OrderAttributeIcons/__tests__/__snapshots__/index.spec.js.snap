// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="attributeCell"
>
  <MBIcon
    className="attributeIcon"
    color="mb_medium_grey"
    icon="gift"
    inline={true}
    title="Gift Order"
  />
  <MBIcon
    className="attributeIcon"
    color="mb_medium_grey"
    icon="vip"
    inline={true}
    title="VIP Order"
  />
  <MBIcon
    className="attributeIcon"
    color="mb_medium_grey"
    icon="scheduled"
    inline={true}
    title="Scheduled Order"
  />
</div>
`;
